components/brave_rewards/resources/tip/components/tipMediaUser.tsx

```diff
@@ -17,40 +17,64 @@ import { getLocale } from '../../../../common/locale'
 interface Props extends RewardsTip.ComponentProps {
   url: string
   publisher: RewardsTip.Publisher
-  tweetMetaData: RewardsTip.TweetMetaData
+  mediaMetaData: RewardsTip.MediaMetaData
 }
 
-class TipTwitterUser extends React.Component<Props, {}> {
+class TipMediaUser extends React.Component<Props, {}> {
 
   get actions () {
     return this.props.actions
   }
 
   onTweet = () => {
-    this.actions.onTweet(
-      `@${this.props.tweetMetaData.screenName}`,
-      this.props.tweetMetaData.tweetId)
+    const mediaMetaData = this.props.mediaMetaData
+    if (!mediaMetaData) {
+      return
+    }
+
+    if (mediaMetaData.mediaType === 'twitter') {
+      this.actions.onTweet(
+        `@${mediaMetaData.screenName}`,
+        mediaMetaData.tweetId)
+    } else if (mediaMetaData.mediaType === 'reddit') {
+      let name = this.props.publisher.name
+      this.actions.onTweet(name, '')
+    }
     this.actions.onCloseDialog()
   }
 
   render () {
     const { finished, error } = this.props.rewardsDonateData
 
     const publisher = this.props.publisher
-    const tweetMetaData = this.props.tweetMetaData
-    const key =
-      tweetMetaData &&
-      tweetMetaData.tweetText &&
-      tweetMetaData.tweetText.length > 0
-      ? 'tweetTipTitle'
-      : 'tweetTipTitleEmpty'
-    publisher.title = getLocale(key, { user: tweetMetaData.screenName })
+    const mediaMetaData = this.props.mediaMetaData
+    if (!mediaMetaData) {
+      return
+    }
+
+    if (mediaMetaData.mediaType === 'twitter') {
+      const key =
+        mediaMetaData &&
```

Comment:
Since this is inside a conditional for `mediaMetaData` can we remove this check?

Reply:
removed